tests/unit/agent_main_test.py

```diff
@@ -579,21 +603,116 @@ def test_no_log_messages_printed_on_config_parse_on_status_stop(self):
                     % (log_level),
                 )
 
-    def _write_mock_config(self):
-        _, tmp_path = tempfile.mkstemp()
+    def test_essential_monitor_fail(self):
+        """
+        This test checks that agent raises an error when some of its monitors, which
+        are configured as 'stop_agent_on_failure', fail.
 
-        config_data = {
-            "api_key": "bar",
-        }
+        """
+        from scalyr_agent.agent_main import ScalyrAgent
+        from scalyr_agent.platform_controller import PlatformController
 
-        with open(tmp_path, "wb") as fp:
-            fp.write(json.dumps(config_data).encode("utf-8"))
+        # 2. MSI install (only intermediate_certs_path check should be skipped)
+        with mock.patch(
+            "scalyr_agent.__scalyr__.INSTALL_TYPE", __scalyr__.PACKAGE_INSTALL
+        ):
 
-        return tmp_path
+            config_data = {
+                "api_key": "0v7dW1EIPpglMcSjGywUdgY9xTNWQP/kas6qHEmiUG5w-",
```

Comment:
I assume this is not a valid api key? If it is, please make sure it's revoked :)